FreeRTOS-Plus/Demo/AWS/Device_Defender_Windows_Simulator/Device_Defender_Demo/DemoTasks/DefenderDemoExample.c

```diff
@@ -0,0 +1,684 @@
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://www.FreeRTOS.org
+ * http://aws.amazon.com/freertos
+ *
+ */
+
+/* Standard includes. */
+#include <stdlib.h>
+#include <stdint.h>
+
+/* Kernel includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+
+/* JSON Library. */
+#include "core_json.h"
+
+/* Device Defender Client Library. */
+#include "defender.h"
+
+/* MQTT operations. */
+#include "mqtt_demo_helpers.h"
+
+/* Metrics collector. */
+#include "metrics_collector.h"
+
+/* Report builder. */
+#include "report_builder.h"
+
+/* Demo config. */
+#include "demo_config.h"
+
+/**
+ * democonfigTHING_NAME is required. Throw compilation error if it is not defined.
+ */
+#ifndef democonfigTHING_NAME
+    #error "Please define democonfigTHING_NAME to the thing name registered with AWS IoT Core in demo_config.h."
+#endif
+
+/**
+ * @brief The length of #democonfigTHING_NAME.
+ */
+#define THING_NAME_LENGTH                 ( ( uint16_t ) ( sizeof( democonfigTHING_NAME ) - 1 ) )
+
+/**
+ * @brief Number of seconds to wait for the response from AWS IoT Device
+ * Defender service.
+ */
+#define DEFENDER_RESPONSE_WAIT_SECONDS    ( 2 )
+
+/**
+ * @brief Status values of the device defender report.
+ */
+typedef enum
+{
+    ReportStatusNotReceived,
+    ReportStatusAccepted,
+    ReportStatusRejected
+} ReportStatus_t;
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief The MQTT context used for MQTT operation.
+ */
+static MQTTContext_t xMqttContext;
+
+/**
+ * @brief The network context used for mbedTLS operation.
+ */
+static NetworkContext_t xNetworkContext;
+
+/**
+ * @brief Static buffer used to hold MQTT messages being sent and received.
+ */
+static uint8_t ucSharedBuffer[ democonfigNETWORK_BUFFER_SIZE ];
+
+/**
+ * @brief Static buffer used to hold MQTT messages being sent and received.
+ */
+static MQTTFixedBuffer_t xBuffer =
+{
+    .pBuffer = ucSharedBuffer,
+    .size    = democonfigNETWORK_BUFFER_SIZE
+};
```

Comment:
This way of initialization is not compatible with C90. Since our library is C90 compatible, it is better for the demos also  to be compatible.